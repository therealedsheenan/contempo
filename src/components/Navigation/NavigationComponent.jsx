// @flow

import React from 'react';
import { pure } from 'recompose';

import { StyledLink, Navigation } from './styles';

const NavigationComponent = () => (
  <Navigation>
    <StyledLink to="/">Home</StyledLink> | <StyledLink to="/style">Styleguide</StyledLink>
  </Navigation>
);

export default pure(NavigationComponent);
