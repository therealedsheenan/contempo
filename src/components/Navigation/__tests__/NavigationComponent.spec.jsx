import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';

// components
import NavigationComponent from '../NavigationComponent';

test('Navigation existence', () => {
  const component = renderer
    .create(
      <Router>
        <NavigationComponent />
      </Router>
    )
    .toJSON();
  expect(component).toMatchSnapshot();
});

test('Link count', () => {
  const component = shallow(<NavigationComponent />);
  expect(component.find('a').root.length).toEqual(1);
});
