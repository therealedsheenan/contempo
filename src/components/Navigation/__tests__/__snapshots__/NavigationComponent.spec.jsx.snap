// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation existence 1`] = `
<nav
  className="sc-bdVaJa hgLOpN"
>
  <a
    className="sc-bwzfXH etlpSL"
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
   | 
  <a
    className="sc-bwzfXH etlpSL"
    href="/style"
    onClick={[Function]}
  >
    Styleguide
  </a>
</nav>
`;
